src/main/java/io/github/townyadvanced/townyresources/controllers/TownResourceDiscoveryController.java

```diff
@@ -49,6 +49,38 @@ public static void discoverNewResource(Resident resident,
 			throw new TownyException(TownyResourcesTranslation.of("msg_err_survey_too_expensive",
                 TownyEconomyHandler.getFormattedBalance(surveyCost), resident.getAccount().getHoldingFormattedBalance()));
 
+		//Pay for the survey
+		resident.getAccount().withdraw(surveyCost, "Cost of resources survey.");
```

Comment:
This seems to not be inside a TownyEconomyHandler.isActive check, and could break on servers that don't have economy enabled.

Reply:
Fixed